framework/src/main/java/org/checkerframework/framework/stub/AnnotationFileElementTypes.java

```diff
@@ -628,15 +627,23 @@ private void parseEnclosingClass(Element e) {
             return;
         }
         String className = getOutermostEnclosingClass(e);
-        if (className == null || className.isEmpty()) {
+        if (className == null || className.isEmpty() || processingClasses.contains(className)) {
```

Comment:
Can you explain the logic for the new check?
Doesn't this still result in `parseEnclosingClass` returning here and then the caller doesn't see any enclosing annotations?

Reply:
It's possible that some declaration annotations are still missing, but I cannot find a test case that produces an issue. I left a comment so it can be revisited in the future.